src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java

```diff
@@ -410,6 +384,50 @@ private JButton getJButtonClose() {
 		return jButtonClose;
 	}
 
+	private JButton getJButtonDelete() {
+		if (jButtonDelete == null) {
+			jButtonDelete = new JButton(MessageBundle.getMessage("angal.common.delete.btn"));
+			jButtonDelete.setMnemonic(MessageBundle.getMnemonic("angal.common.delete.btn.key"));
+			jButtonDelete.setVisible(false);
+			jButtonDelete.addActionListener(actionEvent -> {
+				
+				if (jTableOutcomes.getSelectedRowCount() > 1 ) {
+					MessageDialog.error(this, "angal.medicalstock.pleaseselectonlyonemovement.msg");
+					return;
+				}
+				int selectedRow = jTableOutcomes.getSelectedRow();
+				if (selectedRow == -1) {
+					MessageDialog.error(this, "angal.medicalstock.pleaseselectoucomemovement.msg");
+					return;
+				}
+				MovementWard selectedMovement = (MovementWard)jTableOutcomes.getValueAt(selectedRow, -1);
+				try {
+					MovementWard movWard = movWardBrowserManager.getLastMovementWard(wardSelected);
+					if (movWard.getDate().compareTo(selectedMovement.getDate()) == 0) {
```

Comment:
why comparing the date and not the id?

Reply:
Because the Id is not présent in the outcome Table

Reply:
what do you mean? the table just shows a subset of information, the id should be in the `selectedMovement` object...

Reply:
No. okay I have update the code and use the id like you suggest